crates/solidity/outputs/cargo/tests/src/doc_examples/visitor_api.rs

```diff
@@ -17,40 +16,38 @@ struct ContractCollector {
 }
 
 impl Visitor<Error> for ContractCollector {
-    fn enter_rule(
+    fn rule_enter(
         &mut self,
         node: &Rc<RuleNode>,
-        _path: &Vec<Rc<RuleNode>>,
-        _range: &TextRange,
+        _cursor: &Cursor,
     ) -> Result<VisitorEntryResponse> {
-        if node.kind != RuleKind::ContractDefinition {
-            return Ok(VisitorEntryResponse::StepIn);
+        if node.kind == RuleKind::ContractDefinition {
```

Comment:
nit: I suggest quitting early when possible (like the existing visitor) for readability, and to decrease nesting. But not a blocker at all.

```rust
if !interesting {
  return;
}
do_something();
```

rather than:

```rust
if interesting {
  do_something();
}
```

Reply:
As a functional programming fan, I prefer to have a single control flow, and therefore not use guards as often.